### PR TITLE
[release][codesign] make script python3 compatible

### PR DESCRIPTION
**Pull request by XilaiZhang**
The goal is to have every new bot run with python3 for better scalability, given that python2 has been deprecated. A non deterministic versioning of python could cause confusions among users.

### CODE REVIEW

**Comment by XilaiZhang** [comment]
umm seems like I don't have permissions to add reviewers. would be great if @christopherfujino @jmagman could take a look. Thank you!

File: codesign.py [hunk 1]
```diff
@@ -395,7 +395,8 @@ def run_and_return_output(command):
         command,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE)
-    return proc.stdout.readlines() + proc.stderr.readlines()
+    byteList = proc.stdout.readlines() + proc.stderr.readlines()
+    return [x.decode('utf-8') for x in byteList]
```
Comment (christopherfujino) [inline codesign.py]: what if a unicode character is more than one byte long?

Comment (christopherfujino) [reply codesign.py]: Will passing `text=True` do what we want? https://docs.python.org/3/library/subprocess.html#popen-constructor

Comment (christopherfujino) [reply codesign.py]: sounds like it: https://docs.python.org/3/library/subprocess.html#frequently-used-arguments

Comment (XilaiZhang) [reply codesign.py]: Thanks for the reply! that's true, I will incorporate this change in the next commit!

**Comment by XilaiZhang** [comment]
Newest commit added support for notarytool based on xcode version. Tested locally and remotely. I will update the newly required settings in doc as well.

File: codesign.py [hunk 2]
```diff
@@ -757,6 +795,10 @@ def main(args, bucket_prefix):
 CODESIGN_USERNAME = os.environ['CODESIGN_USERNAME']
```
Comment (christopherfujino) [inline codesign.py:824]: can we delete this?

File: codesign.py [hunk 3]
```diff
@@ -438,9 +439,38 @@ def upload_zip_to_notary(archive_path):
...
+            ', the decision to use notarytool is '+str(isNotaryTool))
+
+        log('Initiating upload of file %s to notary service...' % archive_path)
+        if isNotaryTool:
+            command = [
+            'xcrun',
+            'notarytool',
+            'submit',
+            archive_path,
+            '--wait',
```
Comment (christopherfujino) [inline codesign.py]: now that we're waiting, does this take longer than the previous version?

File: codesign.py (hunk 2 above)
Comment (XilaiZhang) [reply codesign.py:824]: umm This seem to be used in a lot of places. Maybe I am missing something, why would we like to delete it?

**Comment by XilaiZhang** [comment]
Added notarytool support for polling process as well. removed three armv7 files (context https://github.com/flutter/engine/pull/32664/files#diff-a42df9dcaea2b4e4074966cfe07bb92ee27d8972538d58434017c0c937107679R66) I timed multiple runs from start to finish. The time it takes to success is actually short, around 5 minutes.
```
Codesigning & Notarization successful
Success
./codesign.py b41b4de7d6457899f65ac9267cf7f4d0c57bb04f  119.30s user 23.35s system 49% cpu 4:49.08 total
```
 However, I have only observed one status returned from `notary info`, and that is `Accepted`. I have left comment in code regarding this suspicious status.

File: codesign.py [hunk 4]
```diff
@@ -474,41 +505,67 @@ def upload_zip_to_notary(archive_path):
...
         log_and_exit('Unrecognized output from: %s' % ' '.join(command))
 
     status = match.group(1)
-    if status == 'success':
-        return True
-    if status == 'in progress':
-        log('Notarization is still pending...\n')
-        return False
+    if isNotaryTool:
+        if status == 'Accepted': #seems like accpeted is the equivalent of success, if it means in progress it never finishes
```
Comment (XilaiZhang) [inline codesign.py]: seems like accpeted is the equivalent of success, if it means in progress it never finishes

**Comment by XilaiZhang** [comment]
update: I have retrieved the different statuses from notary tool. I am currently working on some errors observed :`The signature does not include a secure timestamp`, `The executable does not have the hardened runtime enabled`, `The binary is not signed` etc.

File: codesign.py (hunk 4 above)
Comment (christopherfujino) [reply codesign.py]: can manually verify these with `spctl`? https://stackoverflow.com/questions/55074168/how-do-i-verify-that-a-macos-pkg-is-notarized#:~:text=The%20notarization%20process%20works%20by,for%20it%20to%20be%20notarized.

Comment (XilaiZhang) [reply codesign.py]: synced with Chris in microkitchen, was caused by some caching issues and was resolved.

**Comment by XilaiZhang** [comment]
just finished testing locally and remotely. The manual code signing should now be complete.

**Comment by christopherfujino** [comment]
> just finished testing locally and remotely. The manual code signing should now be complete.

nice, I will try this out today

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by christopherfujino — CHANGES_REQUESTED** [review]
Ok, per discussion offline maybe this isn't hte version of the script being run in production right now

**Comment by jmagman** [comment]
@christopherfujino this can probably be closed?

**Comment by christopherfujino** [comment]
Yep, this script is deprecated

**Pull request closed without merging**